packages/devtools/lib/src/timeline/timeline.dart

```diff
@@ -155,7 +152,20 @@ class TimelineScreen extends Screen {
         flameChart.attribute('hidden', frame == null);
         eventDetails.attribute('hidden', frame == null);
 
-        flameChart.updateFrameData(frame);
+        if (debugTimeline && frame != null) {
```

Comment:
I assume the slow performance you are seeing is with debugTimeline=false, right?

Reply:
yeah. The debugTimeline flag only affects our trace event processing anyway, not pulling or processing the samples.